pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -66,10 +66,14 @@ func parseConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigContext,
 		parseDatastoreConfig(config, getconfigCtx)
 		// DeviceIoList has some defaults for Usage and UsagePolicy
 		// used by systemAdapters
-		parseDeviceIoListConfig(config, getconfigCtx)
+		physioChanged := parseDeviceIoListConfig(config, getconfigCtx)
 		// Network objects are used for systemAdapters
-		parseNetworkXObjectConfig(config, getconfigCtx)
-		parseSystemAdapterConfig(config, getconfigCtx, false)
+		networksChanged := parseNetworkXObjectConfig(config, getconfigCtx)
+		if physioChanged || networksChanged {
+			parseSystemAdapterConfig(config, getconfigCtx, true)
```

Comment:
Nit.. You can remove the if block and just do:

forceParseSysAdapter = physioChanged || networksChanged
parseSystemAdapterConfig(config, getconfigCtx, forceParseSysAdapter)

Reply:
Sure, that can be done. But having the if block makes code much simpler to read.